src/servers/http_server_v2.cc

```diff
@@ -2145,131 +1924,73 @@ HTTPAPIServerV2::InferRequestClass::FinalizeResponse(
-        const char* shm_region = nullptr;
-        if (!CheckSharedMemoryData(
-                request_output, &shm_region, &offset, &byte_size)) {
-          // Write outputs into json array (if not shared memory)
-          err = WriteDataToJson(
-              output_metadata[i], allocator, const_cast<void*>(base));
-          if (err != nullptr) {
-            break;
-          }
-        }
+        rapidjson::Value params;
+        params.SetObject();
+        params.AddMember("binary_data_size", binary_size_val, allocator);
+        output_metadata[i].AddMember("parameters", params, allocator);
       }
     } else {
-      // Case when output uses classification
-      uint64_t batch_size = 0;
-      err = TRTSERVER2_InferenceRequestOutputClassBatchSize(
-          request, output_name, &batch_size);
-      if (err != nullptr) {
-        break;
-      }
-
-      rapidjson::Value shape_array(rapidjson::kArrayType);
-      shape_array.PushBack(batch_size, allocator);
-      shape_array.PushBack(class_size, allocator);
-      output_metadata[i].AddMember("shape", shape_array, allocator);
-
-      rapidjson::Value datatype_val("STRING");
-      output_metadata[i].AddMember("datatype", datatype_val, allocator);
-
-      std::vector<int32_t> idx(batch_size * class_size);
-      std::vector<float> value(batch_size * class_size);
-      std::vector<char*> label(batch_size * class_size);
-      err = TRTSERVER2_InferenceRequestOutputClasses(
-          request, output_name, idx.data(), value.data(), label.data());
-      if (err != nullptr) {
-        break;
-      }
-
-      std::vector<std::string> class_string(batch_size * class_size);
-      rapidjson::Value batch_class_array(rapidjson::kArrayType);
-      size_t count = 0;
-      for (size_t i = 0; i < batch_size; i++) {
-        rapidjson::Value class_array(rapidjson::kArrayType);
-        for (size_t j = 0; j < class_size; j++) {
-          const char* label_string = label[count];
-          if (strcmp(label_string, "") == 0) {
-            class_string[count] =
-                std::to_string(idx[count]) + ":" + std::to_string(value[count]);
-          } else {
-            class_string[count] = std::to_string(idx[count]) + ":" +
-                                  std::to_string(value[count]) + ":" +
-                                  std::string(label_string);
-          }
-          rapidjson::Value class_str(
-              class_string[count].c_str(), class_string[count].size(),
-              allocator);
-          class_array.PushBack(class_str, allocator);
-          count++;
-        }
-        batch_class_array.PushBack(class_array, allocator);
+      uint64_t offset = 0, byte_size = 0;
+      const char* shm_region = nullptr;
+      if (!CheckSharedMemoryData(
+              request_output, &shm_region, &offset, &byte_size)) {
+        // Write outputs into json array (if not shared memory)
+        WriteDataToJson(output_metadata[i], allocator, const_cast<void*>(base));
```

Comment:
Where is the handling for classification output?

Reply:
It's move inside the server core, the frontend server won't be aware of it and will see cls outputs the same way as regular output.
See discussion
https://github.com/NVIDIA/triton-inference-server/pull/1267#discussion_r403347170